include/arch/arc/v2/arcv2_irq_unit.h

```diff
@@ -98,7 +98,7 @@ static ALWAYS_INLINE
 void _arc_v2_irq_unit_prio_set(int irq, unsigned char prio)
 {
 	_arc_v2_aux_reg_write(_ARC_V2_IRQ_SELECT, irq);
-#ifdef CONFIG_ARC_HAS_SECURE
+#if defined(CONFIG_ARC_HAS_SECURE) && defined(CONFIG_ARC_SECURE_FIRMWARE)
```

Comment:
All interrupts either Secure or Normal is probably too strong, some should be allocated to Secure and some to Normal depending on application. For now ok, but add a note (or issue) to support configuration/allocation of interrupts to Secure or Normal.

Reply:
It requires a mechinism to tell which interrupts are secure, which are normal